tests/unit_tests/test_workflow.py

```diff
@@ -369,3 +370,35 @@ def test_fdr_manager_fit_predict():
     assert fdr_manager_new.get_classifier(FDR_TEST_FEATURES).fitted is True
 
     os.remove(temp_path)
+
+
+def ms2_optimizer_test():
+    test_dict = defaultdict(list)
+    test_dict["var"] = list(range(100))
+
+    ms2_optimizer = optimizer.MS2Optimizer(100)
+
+    assert ms2_optimizer.optimal_tolerance is None
+    assert ms2_optimizer.round == -1
+    assert ms2_optimizer.check_stopping_conditions() is False
+
+    ms2_optimizer.initiate()
```

Comment:
I'm not quite sure why there is a `initiate` function. The naming is a bit ambigous with the object constructor `__init__`.

Would the following work:

```
if not optimizer.has_converged():
   optimizer.step() # or .update()
```

Reply:
There is an initiate function (which I will rename) so that calling the check function multiple times doesn't change the tolerance value. This issue arose during debugging and almost certainly won't be important for the actual use of the object, so I suppose there is no real need to separate the initiation from the check.

Reply:
Just realizing another reason for writing and actually also reviewing unit tests: you can judge much better how the new interfaces work in practise than by just looking at the code

Reply:
Taking this approach will probably resolve the problem (https://github.com/MannLabs/alphadia/pull/283#discussion_r1691073315).